aten/src/ATen/native/cudnn/ConvShared.cpp

```diff
@@ -491,6 +491,77 @@ Tensor cudnn_convolution_transpose_backward_weight(
       padding, stride, dilation, groups, benchmark, deterministic, allow_tf32);
 }
 
+Tensor cudnn_convolution_add_relu(
+    const Tensor& input_t,
+    const Tensor& weight_t,
+    const optional<Tensor>& bias_t,
+    const optional<Tensor>& z_t,
+    optional<Scalar> alpha_t,
+    IntArrayRef padding,
+    IntArrayRef stride,
+    IntArrayRef dilation,
+    int64_t groups) {
+  // Skip all the shape and type checking because frozen-model opt should
+  // catch problems during the compilation
+  TensorArg input{input_t, "input", 1}, weight{weight_t, "weight", 2};
+
+  auto memory_format = cudnn_conv_use_channels_last(*input, *weight)
+      ? at::MemoryFormat::ChannelsLast
+      : at::MemoryFormat::Contiguous;
+  auto output_t = at::native::empty_cuda(
+      conv_output_size(
+          input->sizes(), weight->sizes(), padding, stride, dilation),
+      /*dtype=*/input->scalar_type(),
+      /*layout=*/c10::nullopt,
+      /*device=*/kCUDA,
+      /*pin_memory=*/c10::nullopt,
+      /*memory_format=*/memory_format);
+  if (output_t.numel() == 0) {
+    return output_t;
+  }
+  TensorArg output{output_t, "result", 0};
+
+  // See #4500
+  Tensor weight_contig = weight->contiguous(memory_format);
+  // Make sure that NC11 strides follow formula
+  weight_contig.resize_(weight_contig.sizes(), memory_format);
+  Tensor input_contig = input->contiguous(memory_format);
+  input_contig.resize_(input_contig.sizes(), memory_format);
+
+  Tensor bias_contig;
+  if (bias_t.has_value()) {
+    TensorArg bias{bias_t.value(), "bias", 3};
+    bias_contig = bias->contiguous(memory_format);
+    bias_contig.resize_(bias_contig.sizes(), memory_format);
+  } else {
+    bias_contig = *output;
```

Comment:
you are setting bias to uninitialized output tensor here, how does this work?

Reply:
I thought "auto output_t = at::native::empty_cuda(..." initializes output as zero. If it is not the right assumption, I will create zeros explicitly.

Reply:
It's not the right assumption, but ideally you should avoid explicitly creating tensor of zeros, and instead use corresponding arguments in cudnn call (like beta=0) to make sure those uninitialized values are never read and only written to.